### DISCUSSION TITLE
Fabrication files released

### DISCUSSION THREAD

**Discussion by jcard0na** [cardonabits/haxo-hw#10] [Announcements]
Due to popular demand I have checked in the fabrication files for the Haxophone.  See [this commit](https://github.com/jcard0na/haxo-hw/commit/c86e341e39c9634ad01f00c04da1b016b08e4c1b) and the discussion in #9.

Please note that these were freshly generated from the Kicad project and incorporate some changes I had to make accommodate part shortage.  The good news is that as of today, all the parts I use are in stock at JLCPCB.  The bad news is that these files have not been used to fabricate boards and should be considered "untested prototypes".

If you are brave and interested in building one for yourself, here are some suggestions:

1. Given that JLCPCB minimum order is qty 5 and that unit price drops rapidly with quantity, you can use this forum to pool with others and split an order.  Just start a new discussion "Pool to order haxophones" or something like that...

2. The quote I obtained from JLCPCB to build 5 units is 210 EUR (220 USD) or 42 EUR (45 USD) per board.  Building 15 is 517 EUR (545 USD) is 35 EUR (37 USD) per board.  This is a drop of 20%.

3. Once/if you order, feel free to create an issue and I will assist with board bring up.  I tried to document everything [here](https://github.com/jcard0na/haxo-hw/blob/main/docs/assembly.md), but I'm sure I missed something!

Hack on and be musical!

**Comment by ddodge40** [comment]
Would like to order 2 of them.  Please let me know when others have requested a build.
ddodge40 at Verizon.net
Thanks

**Reply by ddodge40** [reply]
Has anyone else been added to the list for ordering the boards?
Thanks

**Reply by jcard0na** [reply]
Hi @ddodge40,

No, no one else has signed up for this.  Also, I've been told that this is not the best way to handle pre-order type campaigns.  I'm talking to the folks at crowd supply to see if they can help me with the logistics of a small batch of haxophones.  The first response was very positive.  I'll post here if there is progress there.

Thanks!

**Reply by thaytan** [reply]
This was in today's CrowdSupply update email. I hope you get a bunch of people - this looks fun! Looking forward to seeing the CrowdSupply pricing.

**Comment by ddodge40** [comment]
Thanks. My Grandson’s are hopeful 

Don Dodge
Cell 714-271-7888
W6SNX

**Comment by ddodge40** [comment]
Do you have any update on possible orders?

Thanks


> On Jan 23, 2023, at 7:39 AM, Javier Cardona ***@***.***> wrote:
> 
> 
> Hi @ddodge40 <https://github.com/ddodge40>,
> 
> No, no one else has signed up for this. Also, I've been told that this is not the best way to handle pre-order type campaigns. I'm talking to the folks at crowd supply to see if they can help me with the logistics of a small batch of haxophones. The first response was very positive. I'll post here if there is progress there.
> 
> Thanks!
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/jcard0na/haxo-hw/discussions/10#discussioncomment-4758114>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADTPHL7LUPBMIM5PS2Q5B4TWT2Q2RANCNFSM6AAAAAASSJGGTU>.
> You are receiving this because you were mentioned.
> 

Donald Dodge
***@***.***
W6SNX
Cell 714-271-7888

**Comment by jcard0na** [comment]
I'm working with Crowd Supply to sell a batch of haxophones through their
platform.  I still don't have a timeline yet.  I'll let you know!
Javier

**Comment by ddodge40** [comment]
Any news on getting more customers.
Thanks

> On Jan 23, 2023, at 7:39 AM, Javier Cardona ***@***.***> wrote:
> 
> 
> Hi @ddodge40 <https://github.com/ddodge40>,
> 
> No, no one else has signed up for this. Also, I've been told that this is not the best way to handle pre-order type campaigns. I'm talking to the folks at crowd supply to see if they can help me with the logistics of a small batch of haxophones. The first response was very positive. I'll post here if there is progress there.
> 
> Thanks!
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/jcard0na/haxo-hw/discussions/10#discussioncomment-4758114>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADTPHL7LUPBMIM5PS2Q5B4TWT2Q2RANCNFSM6AAAAAASSJGGTU>.
> You are receiving this because you were mentioned.
> 

Donald Dodge
***@***.***
W6SNX
Cell 714-271-7888

**Reply by jcard0na** [reply]
Following up on this... the campaign is not live yet, but the project has been announced publicly: https://www.crowdsupply.com/cardona-bits/haxophone

**Reply by raychmond** [reply]
@jcard0na Thank you for developing and sharing such a beautiful instrument. I'm very interested in building my own. I tried to order the PCB and assembly through JCL, it's too expensive because I will only need one. I saw @ddodge40 also wants to build one. Did you order already or still wait for other people to join? 
Thanks!

**Comment by jcard0na** [comment]
Hi Donald,

As mentioned earlier, I'm working with Crowd Supply to launch a pre-order campaign of haxophones.  My estimate is that we'll launch in September, but it is my first time doing this, so I might be too pessimistic (or optimistic!).
Of course, you could always order directly.  I think JLCPCB's minimum order for prototypes is 2.  The unit cost will be higher, but you could have it much sooner than September.

Best,
Javier

**Comment by ddodge40** [comment]
Have not ordered yet.  Still hoping for others to join the	list.

Thanks

> On Mar 17, 2023, at 8:05 PM, raychmond ***@***.***> wrote:
> 
> 
> @jcard0na <https://github.com/jcard0na> Thank you for developing and sharing such a beautiful instrument. I'm very interested in building my own. I tried to order the PCB and assembly through JCL, it's too expensive because I will only need one. I saw @ddodge40 <https://github.com/ddodge40> also wants to build one. Did you order already or still wait for other people to join?
> Thanks!
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/jcard0na/haxo-hw/discussions/10#discussioncomment-5351194>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADTPHL2DNSG634TQVIKDCTDW4UQ77ANCNFSM6AAAAAASSJGGTU>.
> You are receiving this because you were mentioned.
> 

Donald Dodge
***@***.***
W6SNX
Cell 714-271-7888

**Comment by ddodge40** [comment]
I would like 2 and he wants 1. So do we need 2 more to make 5
Thanks 

Don Dodge
Cell 714-271-7888
W6SNX

**Reply by raychmond** [reply]
Do you want to have 2 assembled? I found that if we order without pressure sensor, the price is much lower not only because of the pressure sensor part price, but also because of the assembly package price. If you don't mind without the pressure sensor, we can make 3 assembled without pressure sensor and 2 without assembly.  The total cost will be manageable, I would say around $60-70 each of us. Certainly, we need to buy a pressure sensor separately and solder on it. It will be an additional $25 + shipping for the pressure sensor. 
What do you think?

**Comment by ddodge40** [comment]
Sounds great   Let’s do it 
Thanks 

Don Dodge
Cell 714-271-7888
W6SNX

**Comment by ddodge40** [comment]
Yes I will buy 2 and also get the pressure sensor. Please let me know when and how to pay for them Thanks 


Sent from the all new AOL app for iOS


On Saturday, March 18, 2023, 6:55 PM, raychmond ***@***.***> wrote:




Do you want to have 2 assembled? I found that if we order without pressure sensor, the price is much lower not only because of the pressure sensor part price, but also because of the assembly package price. If you don't mind without the pressure sensor, we can make 3 assembled without pressure sensor and 2 without assembly. The total cost will be manageable, I would say around $60-70 each of us. Certainly, we need to buy a pressure sensor separately and solder on it. It will be an additional $25 + shipping for the pressure sensor.
What do you think?

**Comment by jcard0na** [comment]
I'll buy one as well, this way I can support you with assembly and help bring down the price a little bit.
@raychmond let me know how to send you the money (Paypal?).
Thanks for taking the initiative on this!

**Reply by raychmond** [reply]
Thanks both of you!
JCL only allows either 2 assembled or 5 assembled. So I will order 5 assembled without pressure sensor. The total cost will be C$129.78 Canadian Dollars including shipping. I already applied a new customer discount of C$13.72. The cost with shipping to my home is C$26 per assembled PCB board without pressure sensor. 
Once I received the boards, I will ship to each of you. I don't know how much the shipping will be for each of you. But I will use Canada Post to send the boards. My guess is around C$20-30 shipping unless you live somewhere that is way too far. After I ship the package, I will let you know the total cost (C$26/board + shipping), so you can pay me through paypal. 
So far, @ddodge40 wants 2 assembled, @jcard0na will have 1 assembled. I will keep 1. So there is an extra one available for anyone who wants.
If you are okay with this plan, I will order today and it will take 2-3 days processing and 8-15 days shipping to my home.

Thanks!

**Reply by jcard0na** [reply]
Sounds good to me as well.  I'm ok to send you the money now so you don't have to front the whole cost of the JLCPCB order.

**Reply by raychmond** [reply]
@jcard0na That's okay. But thank you though. I can handle that. You've done a lot. You don't even ask for a penny.

@ddodge40 we ordered the assembled PCB board without pressure sensor. You will need to order the pressure sensor separately by yourself, or I will try to order some and see if I can ship to you along with the board.

I have placed the order from JCLPCB. I will keep you posted.

Thanks!

**Reply by jcard0na** [reply]
Great!  BTW, other than the pressure sensor, you will need the mechanical and through hole parts as well.  These are itemized [here](https://github.com/jcard0na/haxo-hw#bill-of-materials).  (What you will get from JLCPCB is the first item on that list).

I'm also keeping [the same BOM in octopart](https://octopart.com/bom-tool/1uKwuwiE), which is useful to track the evolution of pricing and availability.  

And you'll need to 3D print the neck and thumb rests.  JLCPCB do offer 3D printing services, but you are probably better off if you find a local supplier:  this way you can really experiment with materials and color to truly personalize your instrument.

**Reply by raychmond** [reply]
Thanks a lot Javier. I've read through your assembly guidance, BOM, and ordered all the required parts. The BOM in octopart is very helpful. @ddodge40 I think you'd better order pressure sensor and other parts yourself to avoid extra shipping from me.

**Reply by raychmond** [reply]
Just got an email from JCL. They said the PCB is 3 designs instead of 1 in my original order. I have paid the additional C$32 charge. So the total cost right now is C$162.17, i.e. C$32.43 per assembled PCB. I think it's still a reasonable price for each of us.

Thanks!

**Reply by jcard0na** [reply]
@raychmond I included the [JLCPCB ordering parameters](https://github.com/jcard0na/haxo-hw/blob/main/builds/jlcpcb-ordering-info.txt) with the build files, not sure if you saw it.

In particular, there's the [3 designs](https://github.com/jcard0na/haxo-hw/blob/main/builds/jlcpcb-ordering-info.txt#L11) requirement, as each of the panels in the haxophone counts a one design.

I also recommend paying for [lead-free HASL surface finish](https://github.com/jcard0na/haxo-hw/blob/main/builds/jlcpcb-ordering-info.txt#L18).  That's just for peace of mind, not really necessary for functionality.

**Reply by raychmond** [reply]
@jcard0na Yes, I saw your file. The problem was if I use 3 designs at the very beginning, it will require the standard package instead of economic package , which is too expensive. So I changed 1. It went through anyway with a good price. Even at the end, they charged this extra C$32, it's still way lower than the standard package (I remember it was C$200+ for 5 assembled boards without pressure sensors.). Anyway, I used the same parameters as yours except this # of design. Now they have approved the product file and in production status.

**Comment by ddodge40** [comment]
Great plan.  Go ahead

Thanks


> On Mar 19, 2023, at 8:18 AM, raychmond ***@***.***> wrote:
> 
> 
> Thanks both of you!
> JCL only allows either 2 assembled or 5 assembled. So I will order 5 assembled without pressure sensor. The total cost will be C$129.78 Canadian Dollars including shipping. I already applied a new customer discount of C$13.72. The cost with shipping to my home is C$26 per assembled PCB board without pressure sensor.
> Once I received the boards, I will ship to each of you. I don't know how much the shipping will be for each of you. But I will use Canada Post to send the boards. My guess is around C$20-30 shipping unless you live somewhere that is way too far. After I ship the package, I will let you know the total cost (C$26/board + shipping), so you can pay me through paypal.
> So far, @ddodge40 <https://github.com/ddodge40> wants 2 assembled, @jcard0na <https://github.com/jcard0na> will have 1 assembled. I will keep 1. So there is an extra one available for anyone who wants.
> If you are okay with this plan, I will order today and it will take 2-3 days processing and 8-15 days shipping to my home.
> 
> Thanks!
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/jcard0na/haxo-hw/discussions/10#discussioncomment-5360117>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADTPHL7TEAMB4EXMZTXXNDTW44PWDANCNFSM6AAAAAASSJGGTU>.
> You are receiving this because you were mentioned.
> 

Donald Dodge
***@***.***
W6SNX
Cell 714-271-7888

**Comment by ddodge40** [comment]
For the pressure sensor, should I order them in the US,  If so , please send me the part number 
Thanks 


Don Dodge
Cell 714-271-7888
W6SNX

**Reply by jcard0na** [reply]
MPXV7007DP
Please see [this message](https://github.com/jcard0na/haxo-hw/discussions/10#discussioncomment-5362394) for links to the full bill of materials of parts you will need to buy and assemble.

**Reply by raychmond** [reply]
I ordered through Newark. The pressure sensor is from their UK warehouse because it's either very expensive or out of stock in Canada. You may find some other suppliers in the States. Javier has a very good document where you can follow through to order parts.

**Comment by ddodge40** [comment]
Was able to order 2 from Amazon. Most other sites were out of stock
Thanks 

Don Dodge
Cell 714-271-7888
W6SNX

**Comment by jcard0na** [comment]
@raychmond Not sure if you already ordered the pressure sensors.  Today I found that you can order up to free five samples of `MPXV7007GC6U` on the [NXP online store](https://store.nxp.com).  I checked the datasheet, and this part should be a drop-in replacement for `MPXV7007DP`.  I've ordered five to test them out.  You might consider getting a few for yourself while the are still available as free samples.

**Reply by raychmond** [reply]
Thank you so much @jcard0na
I ordered one pressure sensor from Newark last week. But I will have an extra PCB. So I ordered another free sample from NXP. It seems I need to order second Pi Zero as well:)
The PCBs from JCL have been shipped yesterday. They should arrive my home 8-15 days later. I will let you know once I receive them.

Thanks!

**Comment by ddodge40** [comment]
My order with Amazon was cancelled. When they went to ship, they could not locate them.  Now have ordered from Digi-Key and expected ship date is 4/38.
Any update on when the boards will arrive?

Thanks

> On Mar 28, 2023, at 2:24 AM, Javier Cardona ***@***.***> wrote:
> 
> 
> @raychmond <https://github.com/raychmond> Not sure if you already ordered the pressure sensors. Today I found that you can order up to free five samples of MPXV7007GC6U on the NXP online store <https://store.nxp.com/>. I checked the datasheet, and this part should be a drop-in replacement for MPXV7007DP. I've ordered five to test them out. You might consider getting a few for yourself while the are still available as free samples.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/jcard0na/haxo-hw/discussions/10#discussioncomment-5451269>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADTPHL5EQXUUBXXA2DSBOYDW6KU4DANCNFSM6AAAAAASSJGGTU>.
> You are receiving this because you were mentioned.
> 

Donald Dodge
***@***.***
W6SNX
Cell 714-271-7888

**Reply by jcard0na** [reply]
Hi @ddodge40 

See [this exchange](https://github.com/jcard0na/haxo-hw/discussions/10#discussioncomment-5451269) for both, a way to get a free pressure sensor sooner than *4/38* (!?), and an update on the boards by @raychmond

**Comment by ddodge40** [comment]
will the single air port NXT sensor work, or do we need both in and out?

Thanks

> On Mar 28, 2023, at 7:49 AM, Javier Cardona ***@***.***> wrote:
> 
> 
> Hi @ddodge40 <https://github.com/ddodge40>
> See this exchange <https://github.com/jcard0na/haxo-hw/discussions/10#discussioncomment-5451269> for both, a way to get a free pressure sensor sooner than 4/38 (!?), and an update on the boards by @raychmond <https://github.com/raychmond>
> —
> Reply to this email directly, view it on GitHub <https://github.com/jcard0na/haxo-hw/discussions/10#discussioncomment-5455221>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADTPHL5VWMJPFB3XIKX3DBDW6L3BDANCNFSM6AAAAAASSJGGTU>.
> You are receiving this because you were mentioned.
> 

Donald Dodge
***@***.***
W6SNX
Cell 714-271-7888

**Reply by jcard0na** [reply]
I believe it should work, but I have not tried it.  

Both sensors measure positive and negative pressures local to the atmospheric pressure.  The second port on the original sensor (MPXV7007DP) can also be used to measure pressure relative to something attached to the second port.  But on the haxophone, it is left unconnected so the behavior of the two should be identical.

**Comment by ddodge40** [comment]
Thanks for the pressure sensor update.  NXP is shipping mine tomorrow.  

I 3D printed the parts and they look great.  

any news on the boards?

 Thanks

> On Mar 28, 2023, at 2:52 PM, Javier Cardona ***@***.***> wrote:
> 
> 
> I believe it should work, but I have not tried it.
> 
> Both sensors measure positive and negative pressures local to the atmospheric pressure. The second port on the original sensor (MPXV7007DP) can also be used to measure pressure relative to something attached to the second port. But on the haxophone, it is left unconnected so the behavior of the two should be identical.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/jcard0na/haxo-hw/discussions/10#discussioncomment-5459130>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADTPHLYE4LDSCVHD43K27Q3W6NMRLANCNFSM6AAAAAASSJGGTU>.
> You are receiving this because you were mentioned.
> 

Donald Dodge
***@***.***
W6SNX
Cell 714-271-7888

**Reply by raychmond** [reply]
The package from JCL has arrived in Canada. I believe I can get it Monday.

**Comment by raychmond** [comment]
I have received the package. Please send your address to my email yaming.luo at yahoo.com. I will ship it asap.

Thanks!

**Comment by raychmond** [comment]
Hi @jcard0na, while I'm assembling the haxophone and 3D printing the parts, I tried to install the haxo-rs on my raspberry pi zero 2. I've got some errors. I'm not sure if I understand correctly or not. This is how I did (headless through SSH).

Thank you!

1. I have installed the latest bullseye 64bit.
2. I installed the Rust and fluidsynth
3. git clone the haxo-rs repo, go to this folder and run "cargo run"
4. I've got some errors:

Compiling haxo001 v0.1.0 (/home/pi_haxo/haxo-rs)
warning: unused borrow that must be used
   --> src/notemap.rs:113:9
    |
113 |         &self.notemap.insert(key, value);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the borrow produces a value
    |
    = note: `#[warn(unused_must_use)]` on by default
help: use `let _ = ...` to ignore the resulting value
    |
113 |         let _ = &self.notemap.insert(key, value);
    |         +++++++

warning: unused borrow that must be used
   --> src/notemap.rs:118:9
    |
118 |         &self.notemap.remove(key);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^ the borrow produces a value
    |
help: use `let _ = ...` to ignore the resulting value
    |
118 |         let _ = &self.notemap.remove(key);
    |         +++++++

error: linking with `cc` failed: exit status: 1
....... (omitted many contents here)

  = note: /usr/bin/ld: cannot find -lfluidsynth
          collect2: error: ld returned 1 exit status
warning: `haxo001` (bin "haxo001") generated 2 warnings
error: could not compile `haxo001` due to previous error; 2 warnings emitted

**Reply by jcard0na** [reply]
Hi @raychmond,

The warnings seem to have cropped up with the newer rust compiler.  They are not blocking progress.  ~If you create an issue with those errors, I'll try to fix them.~
EDIT:  I just pushed a fix for those warnings.  No need to do anything on your end.

The error is caused by a link failure.  haxo-rs is a rust application that links with [libfluidsynth](https://github.com/FluidSynth/fluidsynth/wiki/Download).  You will need to install that separately.

>I installed the Rust and fluidsynth

How die you install fluidsynth?  `apt-get install fluidsynth`?  Is there a libfluidsynth under `/usr/lib` or `/usr/local/lib`?

**Reply by jcard0na** [reply]
This is how this looks on my Raspberry Pi Zero:
```
	pi@raspberrypi-three:~ $ ldd /usr/local/bin/haxo001 | grep fluid
	libfluidsynth.so.1 => /lib/arm-linux-gnueabihf/libfluidsynth.so.1 (0xb6ce3000)
```
and
```
pi@raspberrypi-three:~ $ apt list --installed | grep fluid
fluid-soundfont-gm/stable,now 3.1-5.1 all [installed,automatic]
fluidsynth/stable,now 1.1.11-1 armhf [installed]
libfluidsynth-dev/stable,now 1.1.11-1 armhf [installed]
libfluidsynth1/stable,now 1.1.11-1 armhf [installed,automatic]
```
Seems like you should install `libfluidsynth-dev` as well.  Maybe you can submit a pull request against [haxo-rs](https://github.com/jcard0na/haxo-rs#readme) :wink ?

**Reply by raychmond** [reply]
Thanks Javier!

Yes, I installed fluidsynth by "sudo apt-get install fluidsynth"

I don't see libfluidsynth under /usr/lib or /usr/local/lib

Thanks!

**Reply by raychmond** [reply]
> This is how this looks on my Raspberry Pi Zero:
> 
> ```
> 	pi@raspberrypi-three:~ $ ldd /usr/local/bin/haxo001 | grep fluid
> 	libfluidsynth.so.1 => /lib/arm-linux-gnueabihf/libfluidsynth.so.1 (0xb6ce3000)
> ```
> 
> and
> 
> ```
> pi@raspberrypi-three:~ $ apt list --installed | grep fluid
> fluid-soundfont-gm/stable,now 3.1-5.1 all [installed,automatic]
> fluidsynth/stable,now 1.1.11-1 armhf [installed]
> libfluidsynth-dev/stable,now 1.1.11-1 armhf [installed]
> libfluidsynth1/stable,now 1.1.11-1 armhf [installed,automatic]
> ```
> 
> Seems like you should install `libfluidsynth-dev` as well. Maybe you can submit a pull request against [haxo-rs](https://github.com/jcard0na/haxo-rs#readme) :wink ?

Yes I don't see libfluidsynth-dev. I do see three others installed. As a workaround, how do I install fluidsynth-dev?

**Reply by raychmond** [reply]
This is what I've got by running apt list

pi_haxo@haxo:~ $ apt list --installed |grep fluid

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

fluid-soundfont-gm/stable,now 3.1-5.2 all [installed,automatic]
fluidsynth/stable,now 2.1.7-1.1 armhf [installed]
libfluidsynth2/stable,now 2.1.7-1.1 armhf [installed,automatic]

**Reply by jcard0na** [reply]
```
apt get install libfluidsynth-dev
```
and this is not a workaround, this would be a required package (dependecy) to build `haxo-rs`.  We should document it in the readme.

**Reply by raychmond** [reply]
Yes, I did run that. But this is what I've got:

pi_haxo@haxo:~ $ sudo apt-get install fluidsynth-dev
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package fluidsynth-dev

**Reply by raychmond** [reply]
Is it because I'm using bullseye, not buster? Sorry Javier, I'm not a developer. I may have asked some dummy questions. Thank you so much for your help!

**Reply by jcard0na** [reply]
No worries.  Let me investigate, I'm not familiar with bullseye

**Reply by jcard0na** [reply]
duh, my bad, the package name is **lib**fluidsynth-dev
Edited my earlier response to avoid confusions.

**Reply by jcard0na** [reply]
And it's listed here: https://www.raspberryconnect.com/raspbian-packages/77-raspbian-libdevel
Exists for both buster and bullseye

**Reply by raychmond** [reply]
Thanks a lot! I have installed the libfluidsynth-dev successfully. I will re-compile the haxo-rs.

**Reply by raychmond** [reply]
Hi Javier,

I was able to compile on my pi zero 2. However, when it started the haxophone, it immediately exited because the GPIO scan failed. I ran the test program. It failed as well. 
I then used my pi 4b. It compiled successfully. But it failed somehow.  Please see below error message:

pi_haxo@haxo:~/haxo-rs $ cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 1.10s
     Running `target/debug/haxo001`
fluidsynth: warning: Failed to set thread to high priority
Synth created
Starting haxophone...
fluidsynth: warning: Failed to allocate a synthesis process. (chan=0,key=59)
fluidsynth: warning: Failed to allocate a synthesis process. (chan=0,key=59)
fluidsynth: warning: Failed to allocate a synthesis process. (chan=0,key=59)

I then ran the test. In the all_keys test, I found that only 18 keys have detected. 

Are these two issues related? I mean fluidsynth warning and some keys are not responded?

Thank you!

**Comment by ddodge40** [comment]
I do not think the pi zero can do 64 bit. The pi 4’s can with many issues on old programs 


Sent from the all new AOL app for iOS


On Sunday, April 2, 2023, 9:19 AM, raychmond ***@***.***> wrote:




Hi @jcard0na, while I'm assembling the haxophone and 3D printing the parts, I tried to install the haxo-rs on my raspberry pi zero 2. I've got some errors. I'm not sure if I understand correctly or not. This is how I did (headless through SSH).

Thank you!
   
   - I have installed the latest bullseye 64bit.
   - I installed the Rust and fluidsynth
   - git clone the haxo-rs repo, go to this folder and run "cargo run"
   - I've got some errors:

Compiling haxo001 v0.1.0 (/home/pi_haxo/haxo-rs)
warning: unused borrow that must be used
--> src/notemap.rs:113:9
|
113 | &self.notemap.insert(key, value);
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the borrow produces a value
|
= note: #[warn(unused_must_use)] on by default
help: use let _ = ... to ignore the resulting value
|
113 | let _ = &self.notemap.insert(key, value);
| +++++++

warning: unused borrow that must be used
--> src/notemap.rs:118:9
|
118 | &self.notemap.remove(key);
| ^^^^^^^^^^^^^^^^^^^^^^^^^ the borrow produces a value
|
help: use let _ = ... to ignore the resulting value
|
118 | let _ = &self.notemap.remove(key);
| +++++++

error: linking with cc failed: exit status: 1
....... (omitted many contents here)

= note: /usr/bin/ld: cannot find -lfluidsynth
collect2: error: ld returned 1 exit status
warning: haxo001 (bin "haxo001") generated 2 warnings
error: could not compile haxo001 due to previous error; 2 warnings emitted

**Reply by raychmond** [reply]
Thanks Don! Yeah, I'm now trying the 32bit bullseye. 
BTW, I've got your address. I will ship 2 asap.

Thanks!

**Reply by raychmond** [reply]
I've got the same errors using the 32 bit bullseye. I probably didn't do it right. Can anyone help me?
I've completed the hardware, but stuck at the software part.

**Comment by ddodge40** [comment]
i will try it also.  There are some issues with bullseye and older compilations.  See the Ham radio site km4ack for some hints on how to correct some installation problems.

Is the source code for halo available?  You may need to recompile with the latest python updates.

Thanks


> On Apr 2, 2023, at 10:52 AM, raychmond ***@***.***> wrote:
> 
> 
> I've got the same errors using the 32 bit bullseye. I probably didn't do it right. Can anyone help me?
> I've completed the hardware, but stuck at the software part.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/jcard0na/haxo-hw/discussions/10#discussioncomment-5502965>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADTPHL6C7DSIZFJGHVT7SR3W7G4FHANCNFSM6AAAAAASSJGGTU>.
> You are receiving this because you were mentioned.
> 

Donald Dodge
***@***.***
W6SNX
Cell 714-271-7888

**Comment by ddodge40** [comment]
Where can I find the code haxo-rs file. GitHub said not found. I can see the other .rs files Thanks


Sent from the all new AOL app for iOS


On Sunday, April 2, 2023, 1:52 PM, raychmond ***@***.***> wrote:




Thanks a lot! I have installed the libfluidsynth-dev successfully. I will re-compile the haxo-rs.

**Reply by jcard0na** [reply]
There is no file named `haxo-rs`, it's the name of the repository: https://github.com/jcard0na/haxo-rs

**Comment by ddodge40** [comment]
Looks like I was able to compile. It ended with error “could not find pressure sensor “. As I do not have an assembled unit, maybe that is okI am using a pi zeroThai


Sent from the all new AOL app for iOS


On Sunday, April 2, 2023, 1:49 PM, Javier Cardona ***@***.***> wrote:




And it's listed here: https://www.raspberryconnect.com/raspbian-packages/77-raspbian-libdevel
Exists for both buster and bullseye

**Reply by jcard0na** [reply]
Yes, sounds like you are on the good track.  The initialization of the pressure sensor will fail if no sensor is present: https://github.com/jcard0na/haxo-rs/blob/main/src/main.rs#L71

**Reply by raychmond** [reply]
Hi Javier,

Can you please let me know your address by emailing to yaming.luo at yahoo.com?
I will ship 1 PCB to you today as well as 2 to Don.

I hope all PCBs are alright. I may have soldering problems on some keys because they are not detected by the test program.

Thanks!

**Reply by jcard0na** [reply]
Hi @raychmond,

I sent you my postal address yesterday.  Have you not received it?

If you post test logs I can try to help troubleshoot the failing keys.  Probably best to [file an issue on haxo-rs](https://github.com/jcard0na/haxo-rs/issues) and move the conversation there.  We've stretched this discussion thread a bit already... 😉 

Cheers,

**Reply by raychmond** [reply]
@jcard0na 
Sorry, found your email in the spam folder. 
Great, I will ship the boards this afternoon. After I ship them, I will send your guys the shipping cost plus the board cost (it's 162.17/5=$Canadian Dollar 32.43 per board). I will let you know my paypal account in the email.
Thanks!

**Comment by raychmond** [comment]
Shame on me. I was too confident on my soldering skills. After re-soldering these faulty keys, it passed all test. It's now the problem with the fluidsynth warnings: 
fluidsynth warning: Failed to set thread to high priority
fluidsynth warning: Failed to allocate a synthesis process. (chan=0,key=59)

@jcard0na Can you please see if my setup is right:
I don't do the overlay you recommended. I cloned your git, compile and run on the same SD card as the root system. If this is the problem, should I mount a USB drive and clone, compile and run your programs there?

Thank you so much!

**Reply by jcard0na** [reply]
That's good news.  I'd love to see a picture of the assembled device at some point.

The overlay is just to protect the SD card filesystem integrity.  SD cards are really not meant to be written to so frequently, and they can be easily corrupted if power is disconnected during a write operation.  This is why I recommend the overlay, which essentially makes the SD card read only.  Of course, this is not necessary during initial bring up and testing:  just be sure you `sudo halt` before disconnecting the Raspberry Pi.

Do you get any sound at all?  The warnings are not critical, and might go away if you run `haxo001` as root.

**Reply by raychmond** [reply]
Thank you Javier!

No sound so far. The warning of "Failed to allocate a synthesis process. (chan=0,key=59)" keeps showing up.

How do I run haxo001 as root by the way? 

Thanks!

**Reply by jcard0na** [reply]
To run as root, `sudo haxo001`.  But that is not required to get sound, it just helps run the process at higher priority.

As for not getting sound, let me check some of the basics...

1. Did you connect a speaker or headphone to the minijack on the haxo PCB?  (I'm not sure if you are using a Raspberry 4 which also has its own audio port)
2. Any specific changes you did via `raspi-config`?
3. Any warnings from `haxo001` standard output (other than the two you mentioned)?

**Comment by ddodge40** [comment]
I have my board completed and get the Fluidsynth errors as others. I can play sounds from the board with Qsynth. But no sounds from the haxophone. The cargo tests runs and skips the key press part. Tried several ways to get the test to step through without success.  Looks like a Fluidynth issue
Thanks

**Comment by joshyboy927** [comment]
Hi there, 

Wanted to grab two of these and am also so lost with the PCB ordering process so if anyone wanted to get the other 3 or 2 I can get 3 if needed, but also help with the ordering that would be amazing!

**Reply by nh2** [reply]
@joshyboy927 I would order 2, I'm in Switzerland.

I also just ordered 1 from the Crowd Supply campaign, but would be interested in these separate ones for cheaper and self-assembly.

If it's going well with the self-assembly, I'd also like to offer a course to some friends where we build and play multiple ones.

**Comment by joshyboy927** [comment]
Also a discord server for this project would be amazing

**Comment by nh2** [comment]
@jcard0na You should really put the Crowd Supply campaign prominently into the README, I've been eyeing the Haxophone for long but was intimidated by getting the parts, and only found the campaign by accident (I just ordered one).

**Reply by jcard0na** [reply]
Hi Niklas,

Thanks for the suggestion announcement suggestion and for supporting the project.  I just updated the README.

> I'd also like to offer a course to some friends where we build and play multiple ones.

Please don't hesitate to reach out if you need help to prepare your course.

Best,

**Comment by JinRecords** [comment]
Good day, I've successfully built the Haxophone. However, I do have 4 assembled PCBs left over. I'm selling each PCB for $40 (shipping not included). I live in Asia, so shipping will be more expensive depending on where you live. Please feel free to contact me on Discord: @jinrec for any questions and enquires.
![IMG_20240417_012402](https://github.com/cardonabits/haxo-hw/assets/45748971/bed5e918-1e1d-4234-95ca-3c3cfafa83db)
